packages/loot-core/src/server/accounts/sync.ts

```diff
@@ -537,6 +537,20 @@ export async function reconcileGoCardlessTransactions(acctId, transactions) {
   await createNewPayees(payeesToCreate, [...added, ...updated]);
   await batchUpdateTransactions({ added, updated });
 
+  // Make sure the "cleared" flag is synced up with the parent
+  // transactions
+  let clearedRows = await db.all(`
+    SELECT t.id, p.cleared FROM v_transactions t
+    LEFT JOIN v_transactions p ON t.parent_id = p.id
+    WHERE t.is_child = 1 AND t.cleared != p.cleared
+  `);
+
+  let updatedClearedRows = clearedRows.map(row => ({
+    id: row.id,
+    cleared: row.cleared === 1,
+  }));
+  await batchUpdateTransactions({ updated: updatedClearedRows });
+
```

Comment:
Having to query every transaction like this seems very inefficient.  I think the following would be a better solution:

```diff
diff --git a/packages/loot-core/src/server/accounts/sync.ts b/packages/loot-core/src/server/accounts/sync.ts
index 4733dc6f..4cbea6d5 100644
--- a/packages/loot-core/src/server/accounts/sync.ts
+++ b/packages/loot-core/src/server/accounts/sync.ts
@@ -431,7 +431,7 @@ export async function reconcileGoCardlessTransactions(acctId, transactions) {
       // matched transaction. See the final pass below for the needed
       // fields.
       fuzzyDataset = await db.all(
-        `SELECT id, date, imported_id, payee, category, notes FROM v_transactions
+        `SELECT id, is_parent, date, imported_id, payee, category, notes FROM v_transactions
            WHERE date >= ? AND date <= ? AND amount = ? AND account = ? AND is_child = 0`,
         [
           db.toDateRepr(monthUtils.subDays(trans.date, 4)),
@@ -509,6 +509,17 @@ export async function reconcileGoCardlessTransactions(acctId, transactions) {
       if (hasFieldsChanged(existing, updates, Object.keys(updates))) {
         updated.push({ id: existing.id, ...updates });
       }
+
+      if (existing.is_parent && existing.cleared !== updates.cleared) {
+        const children = await db.all(
+          'SELECT * FROM v_transactions WHERE parent_id = ?',
+          [existing.id],
+        );
+        for (const child of children) {
+          updated.push({ id: child.id, cleared: updates.cleared });
+        }
+      }
     } else {
       // Insert a new transaction
       let finalTransaction = {
```

Reply:
I agree. Thanks for for the suggestion

I've updated it in both the reconcileGoCardlessTransactions and reconcileTransactions functions

Please review